lib/aerospike/client.rb

```diff
@@ -731,6 +730,59 @@ def query(statement, options = nil)
       query_partitions(Aerospike::PartitionFilter.all, statement, options)
     end
 
+    #----------------------------------------------------------
+    # Query/Execute (Supported by Aerospike 3+ servers only)
+    #----------------------------------------------------------
+
+    # QueryExecute applies operations on records that match the statement filter.
+    # Records are not returned to the client.
+    # This asynchronous server call will return before the command is complete.
+    # The user can optionally wait for command completion by using the returned
+    # ExecuteTask instance.
+    #
+    # This method is only supported by Aerospike 3+ servers.
+    # If the policy is nil, the default relevant policy will be used.
+    #
+    # @param statement [Aerospike::Statement] The query or batch read statement.
+    # @param operations [Array<Aerospike::Operation>] An optional list of operations.
+    # @param options [Hash] An optional hash of policy options.
+    # @return [Aerospike::ExecuteTask] An ExecuteTask instance that can be used to wait for command completion.
+    #
+    # @raise [Aerospike::Exceptions::Aerospike] if an error occurs during the operation.
+    def query_execute(statement, operations = [], options = nil)
+      policy = create_policy(options, WritePolicy, default_write_policy)
```

Comment:
I'd rather see two policies here: One for WritePolicy and one for QueryPolicy. That way you'll not have to put new field in WritePolicy that are only related and applicable to the query_execute command.

Reply:
moved records_per_second to statement similar to the java client.